### ISSUE TITLE
Multi-class text classification with one training example per class

### ISSUE THREAD

**Issue by jabzer-research** [konfuzio-ai/document-ai-python-sdk#25]
I am trying to solve a multi-class single-label document classification problem assigning a single class to a document. Documents are domain-specific technical documents, with technical terms:

Train: I have 19 classes with a single document in each class.
Target: I have 77 documents without labels I want to classify to the 19 known classes.
Documents have between 60-3000 tokens after pre-processing.
My entire corpus (19+77 documents) have 65k terms (uni/bi/tri-grams) with 4.5k terms in common (between train and target)

Currently, I am vectorizing documents using a tf-idf vectorizer and reducing dimensions to common terms. Then doing a cosine similarity between train and target.

I am wondering if there is a better way? I cannot use sklearn classifiers due to a single document in each class in train. Any ideas on a possible improvement/direction? Especially:

Does it make sense to use word-embeddings/doc2vec given the small corpus?
Does it make sense to generate synthetic train data from the terms in the training set?
Any other ideas?

https://stackoverflow.com/questions/66982422/multi-class-text-classification-with-one-training-example-per-class

**Comment by al-konfuzio** [comment]
We are working to make the categorization of documents also available through the SDK. Meanwhile, you can try it out using the Konfuzio app.
